src/gpu/utils.cu

```diff
@@ -29,11 +29,22 @@ extern "C" {
     }  
     return(0);
   }
-
   #ifdef __cplusplus
 }
 #endif
 
+#ifdef __cplusplus
+extern "C" {
+  #endif
+
+  int cudaresetdevice_bare(void) {
+    cudaDeviceReset();
```

Comment:
Should we do a CUDACHECK() here?

Reply:
No, can't, because that would fail as no cuda context or it would regenerate the cuda context.  We want to destroy it and be done.

Reply:
@pseudotensor ah I see, so in `cudaresetdevice` we can do it because we are doing an explicit `cudaSetDevice`? Could we add a comment here so I don't add any checks in the future by accident? (during last refactoring I noticed we have a few cuda check macros/methods - would be good to unify that).